erpnext/buying/doctype/supplier/supplier.py

```diff
@@ -51,6 +51,23 @@ def validate(self):
 		validate_party_accounts(self)
 		self.validate_internal_supplier()
 
+	@frappe.whitelist()
+	def get_supplier_group_details(self):
+		doc = frappe.get_doc('Supplier Group', self.supplier_group)
+		self.payment_terms = ""
+		self.accounts = []
+
+		if not self.accounts and doc.accounts:
+			for account in doc.accounts:
+				child = self.append('accounts')
+				child.company = account.company
+				child.account = account.account
+			self.save()
+
+		if not self.payment_terms and doc.payment_terms:
+			self.payment_terms = doc.payment_terms
+
+
```

Comment:
You're already resetting `self.payment_terms` and `self.accounts`, no need to check again if it's empty. Save once at the end.
```suggestion
		if doc.accounts:
			for account in doc.accounts:
				child = self.append('accounts')
				child.company = account.company
				child.account = account.account

		if doc.payment_terms:
			self.payment_terms = doc.payment_terms
        
        self.save()
```